packages/core/src/getPathFromState.tsx

```diff
@@ -180,19 +180,19 @@ export default function getPathFromState<ParamList extends {}>(
         .map((p) => {
           const name = getParamName(p);
 
-          // We don't know what to show for wildcard patterns
-          // Showing the route name seems ok, though whatever we show here will be incorrect
-          // Since the page doesn't actually exist
+          // This can occur when a wildcard matches all routes after a given segment
+          // and the provided path was `/`.
           if (p === '*') {
-            return route.name;
+            return route.path ?? '';
```

Comment:
Let's leave `route.name` for now
```suggestion
            return route.path ?? route.name;
```